### PR TITLE
Add git repository and edit urls to user guide

### PR DESCRIPTION
**Pull request by messense**
https://rust-lang.github.io/mdBook/format/configuration/renderers.html#html-renderer-options

image

### CODE REVIEW

File: guide/book.toml [hunk 1]
```diff
@@ -5,3 +5,7 @@ author = "PyO3 Project and Contributors"
 
 [preprocessor.pyo3_version]
 command = "python3 guide/pyo3_version.py"
+
+[output.html]
+git-repository-url = "https://github.com/PyO3/pyo3/tree/main/guide"
```
Comment (adamreichold) [inline guide/book.toml:10]: Is it customary to link into the tree here instead of just linking the PyO3 repository?

Comment (messense) [reply guide/book.toml:10]: mdbook also link into the tree: https://github.com/rust-lang/mdBook/blob/c8db0c8ec65e9ff8e82c580806a7c5dc03774352/guide/book.toml#L13

**Comment by Psykopear** [comment]
Sorry, opened a PR with the fix: #2636

**Review by davidhewitt — APPROVED** [review]
Great idea!

CI failure looks related to `maturin-action`?

**Comment by messense** [comment]
Rust 1.64.0 requires GLIBC 2.17+, so we need to change `manylinux: auto` in `ci.yml` to `manylinux: 2014`.

**Comment by davidhewitt** [comment]
Thanks. I also pushed #2637 for the other pieces here

**Pull request merged**